### PR TITLE
Separate Evaluation Script

### PR DESCRIPTION
**Pull request by juanmc2005**
## Biggest Changes
- Added Evaluation Experiment classes for speaker verification and sts to abstract away all the ugly configuration needed to test a model
- Added a new `eval.py` script that only tests a saved model
- Cleaned sts baseline code
- Added (but still not using) forward modes for the sts baseline model. The idea is to put this logic in strategy-like objects so it's cleaner and less hard-coded

### CODE REVIEW

File: eval.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+elif args.task == 'sts':
+    experiment = SemEvalModelEvaluationExperiment(model_path=args.model,
+                                                  nfeat=500,
+                                                  data_path=args.sts_path,
+                                                  word2vec_path=args.word2vec,
+                                                  vocab_path=args.vocab,
+                                                  distance=distance,
+                                                  log_interval=args.log_interval,
+                                                  batch_size=args.batch_size)
+    result_format = "[{partition} Spearman = {metric}]"
```
Comment (juanmc2005) [inline eval.py:43]: The name metric could be changed instead of the whole format string

File: losses/base.py [hunk 2]
```diff
@@ -184,10 +186,33 @@ def __call__(self, x, y):
...
+            if model.loss_module is not None:
+                model.loss_module.load_state_dict(checkpoint['loss_module_state_dict'])
+            optimizer.load_state_dict(checkpoint['optim_state_dict'])
+        print(f"Recovered Model. Epoch {checkpoint['epoch']}. Test Metric {checkpoint['accuracy']}")
+        return checkpoint
+
+    def load(self, model: SimNet, path: str):
+        checkpoint = torch.load(path)
+        model.load_common_state_dict(checkpoint['common_state_dict'])
+        if self.loss_name == checkpoint['trained_loss'] and model.loss_module is not None:
```
Comment (juanmc2005) [inline losses/base.py:208]: Since this method is only used for loading prediction models, maybe we can forget about the loss module.
All evaluation methods will use the final embeddings instead of the loss module predictions

File: losses/base.py [hunk 3]
```diff
@@ -196,20 +221,13 @@ def __init__(self, loss_name, model, device, loss_fn, loader, optim, recover=Non
         self.recover = recover
         self.batch_transforms = batch_transforms
         self.callbacks = callbacks
+        self.model_loader = ModelLoader(loss_name)
```
Comment (juanmc2005) [inline losses/base.py:224]: If the loss name is not unsed anywhere else in the class, maybe we can merge `loss_name` and `recover` into a single `model_loader` parameter.
I need to check if this is possible.
